code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -1120,24 +1108,43 @@
 /obj/structure/closet/crate/necropolis/colossus
 	name = "colossus chest"
 
-/obj/structure/closet/crate/necropolis/colossus/bullet_act(obj/item/projectile/P)
-	if(istype(P, /obj/item/projectile/colossus))
-		return BULLET_ACT_FORCE_PIERCE
-	return ..()
+/obj/structure/closet/crate/necropolis/colossus/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_PARENT_ATTACKBY, .proc/try_spawn_loot)
```

Comment:
This seems to be repeated a lot, couldn't chest opening be moved up to /obj/structure/closet/crate/necropoli and then have try_spawn_loot do nothing unless oevrriden.

Reply:
I'm not actually sure. 
I feel like if I moved it up to `/obj/structure/closet/crate/necropolis` it would be try to run `.proc/try_spawn_loot` on that instead of the child object specific to each boss. 

I'll give it a try though.

Reply:
Resolved this with https://github.com/BeeStation/BeeStation-Hornet/pull/6812/commits/96e94e41453625b4093ddc8f5b6f12e0be55e44c